spec-tests/src/main/java/org/candlepin/spec/bootstrap/data/util/CertificateUnpackUtil.java

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright (c) 2009 - 2022 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package org.candlepin.spec.bootstrap.data.util;
+
+import org.candlepin.spec.bootstrap.client.cert.X509Cert;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.bouncycastle.asn1.ASN1Encodable;
+import org.bouncycastle.asn1.ASN1InputStream;
+import org.bouncycastle.asn1.ASN1Primitive;
+import org.bouncycastle.asn1.DEROctetString;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.security.cert.X509Certificate;
+import java.util.Base64;
+import java.util.zip.InflaterInputStream;
+
+
+public final class CertificateUnpackUtil {
```

Comment:
I added a CertificateUtil class which has a extractEntitlementCertificatesFromPayload method that uses some of the logic of these methods like extractPayload() and decompress(). I think that we should combine these two utility classes so that we are not duplicating code and having two different classes doing the same thing,

https://github.com/candlepin/candlepin/blob/master/spec-tests/src/main/java/org/candlepin/spec/bootstrap/data/util/CertificateUtil.java